mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+                `MCM_COUCHDB_URL`.
+            mcm_couchdb_credential (str): Credentials to write data in
+                CouchDB. If it is not provided, it will be tried to retrieve it from
+                `COUCH_CRED`
+            mcm_couchdb_lucene_url (str): McM search engine URL.
+                If it is not provided, it will be tried to retrieve it from
+                `MCM_LUCENE_URL`.
+            mcm_application_url (str): McM web application URL.
+                If it is not provided, it will be tried to retrieve it from
+                - Host: MCM_HOST
+                - Port: MCM_PORT
+            mockup (bool): If True, the creation/validation of mock users
+                will not be perform
+        Raises:
+            RuntimeError: In case some of the required attributes finish to be empty.
+        """
+        self.mcm_couchdb_url = mcm_couchdb_url or os.getenv("MCM_COUCHDB_URL", "")
+        self.mcm_couchdb_credential = mcm_couchdb_credential or os.getenv(
+            "COUCH_CRED", ""
+        )
+        self.mcm_couchdb_lucene_url = mcm_couchdb_lucene_url or os.getenv(
+            "MCM_LUCENE_URL", ""
+        )
+        self.mcm_application_url = mcm_application_url
+
+        # Check the attributes are properly set
+        error_msg: list[str] = []
+
+        if not self.mcm_couchdb_url:
+            error_msg += [
+                "Please the CouchDB connection url via constructor args or `MCM_COUCHDB_URL` variable"
+            ]
+        if not self.mcm_couchdb_credential:
+            error_msg += [
+                "Please set the CouchDB basic credentials via constructor args or `COUCH_CRED` variable"
+            ]
+        if not self.mcm_couchdb_lucene_url:
+            error_msg += [
+                "Please set the CouchDB Lucene connection url via constructor args or `MCM_LUCENE_URL` variable"
+            ]
+        if not self.mcm_application_url:
+            # Check if it is possible to form it from environment
+            # variables.
+            host_from_env: str = os.getenv("MCM_HOST", "")
+            port_from_env: str = os.getenv("MCM_PORT", "")
+            if host_from_env and port_from_env:
+                self.mcm_application_url = f"http://{host_from_env}:{port_from_env}/"
+            else:
+                error_msg += [
+                    (
+                        "Please set the McM application url via constructor args "
+                        "or set the enviroment variables `MCM_HOST` and `MCM_PORT`"
+                    )
+                ]
+
+        # Check and raise
+        if error_msg:
+            compress_msg = "\n".join(error_msg)
+            raise RuntimeError(compress_msg)
+
+        # Check the required mock users are available
+        # Otherwise, create them
+        if not mockup:
+            if not self.check_test_users():
+                self.__include_test_users()
+
+    def __http_request(
```

Comment:
All this method does on top of `requests.request()` is creating the full URL. Remove it.

Reply:
Agree, already solved